internal/metrics/otelmetrics/factory_test.go

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package otelmetrics_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/otel/exporters/prometheus"
+	"go.opentelemetry.io/otel/metric/noop"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+
+	"github.com/jaegertracing/jaeger/internal/metrics/otelmetrics"
+	"github.com/jaegertracing/jaeger/pkg/metrics"
+)
+
+func newTestFactory() metrics.Factory {
+	exporter, _ := prometheus.New()
+	meterProvider := sdkmetric.NewMeterProvider(sdkmetric.WithReader(exporter))
+	return otelmetrics.NewFactory(meterProvider)
+}
+
+func TestCounter(t *testing.T) {
+	factory := newTestFactory()
+	counter := factory.Counter(metrics.Options{
+		Name: "test_counter",
+		Tags: map[string]string{"tag1": "value1"},
+	})
+	require.NotNil(t, counter)
+	counter.Inc(1)
```

Comment:
this needs to test that the corresponding Prometheus value was bumped. See examples in internal/metrics/prometheus/factory_test.go

Reply:
I am stuck. I am not able to figure out  how will I check if the value was bumped if I am creating an OTEL Factory, is there any specific function to check the value and compare it?

Reply:
https://github.com/jaegertracing/jaeger/blob/d33515e2706c5766807ca082fc932658567feeaa/internal/metrics/prometheus/factory_test.go#L49

Reply:
Yes, but aren't all these function definitions local to Prometheus factory only?

Reply:
PTAL.